### DISCUSSION TITLE
Can't update Streamlink, stuck at 2.3.0

### DISCUSSION THREAD

**Discussion by aarony1993** [streamlink/streamlink#5218] [Q&A] [answered]
### Description

Hello everyone.

I haven't used Streamlink in a long time. I tried to update to a current version, but when I checked my version with streamlink --version, I got "streamlink 2.3.0"

I uninstalled and re-installed everything, but I don't know how to get rid of this old installation of 2.3.0

[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.8.7
[cli][debug] Streamlink: 2.3.0
[cli][debug] Requests(2.25.1), Socks(1.7.1), Websocket(1.1.0)
[cli][debug] Arguments:
[cli][debug]  --loglevel=debug
[cli][debug]  --player="C:\Program Files\VideoLAN\VLC\vlc.exe" --file-caching=5000
[cli][debug]  --rtmpdump=C:\Program Files (x86)\Streamlink\rtmpdump\rtmpdump.exe
[cli][debug]  --ffmpeg-ffmpeg=O:\Streamlink\ffmpeg\ffmpeg.exe

**Comment by bastimeyer** [comment, marked as answer]
> [cli][debug] OS: Windows 10
> I uninstalled and re-installed everything

You did not. Your command line shell is still resolving `streamlink` to an old installation because its path is higher up in the `PATH` environment variable, so it gets resolved first.

If you're using cmd.exe / Windows batch, then run `where streamlink` to see the full path of the `streamlink.exe` executable or run `echo %PATH%` to see all the paths where executables get resolved from when not specifying a relative or absolute path. The old Streamlink install dir should be included there.

If you're using powershell, run `(Get-Command streamlink).Path` or `echo $env:PATH`.

https://www.google.com/search?q=windows+path+environment+variable

Find the old install and remove it. Alternatively, specify a relative or absolute path when running streamlink, like `.\streamlink.exe` from the current dir, or `C:\path\to\streamlink.exe`.

> [cli][debug] --rtmpdump=C:\Program Files (x86)\Streamlink\rtmpdump\rtmpdump.exe

You're also still using an invalid config file. `--rtmpdump` and `--rtmp-rtmpdump` have been removed ages ago in 3.0.0. Streamlink >=3.0.0 will raise an error when trying to load a config file with those parameters.
https://streamlink.github.io/cli/config.html

The Windows installer for Streamlink will only try to remove `--rtmp-rtmpdump` from your config file though.
https://github.com/streamlink/windows-builds/blob/master/CHANGELOG.md#400-1-2022-05-01
https://github.com/streamlink/windows-builds/blob/01ec1aaffe776aef24b8743b4f6dbfbd0bd5a00f/installer.nsi#L70
